src/seq/index.rs

```diff
@@ -327,14 +328,14 @@ where
     }
     impl<N> PartialOrd for Element<N> {
         fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
-            self.key.partial_cmp(&other.key)
+            Some(self.cmp(other))
         }
     }
     impl<N> Ord for Element<N> {
         fn cmp(&self, other: &Self) -> core::cmp::Ordering {
-             // partial_cmp will always produce a value,
-             // because we check that the weights are not nan
-            self.partial_cmp(other).unwrap()
+            // partial_cmp will always produce a value,
+            // because we check that the weights are not nan
+            self.key.partial_cmp(&other.key).unwrap()
```

Comment:
Note changes in these `PartialOrd` and `Ord` impls.

Reply:
Yeah, lets not mix up reformatting and other stuff...

Reply:
This is relevant to Clippy lint which insists on using canonical implementation for `PartialOrd`.